addon/bootstrap.js

```diff
@@ -109,13 +114,13 @@ async function startup(addonData, reason) {
   * studyUtils._isEnding means this is a '2nd shutdown'.
   */
 function shutdown(addonData, reason) {
-  console.log("shutdown", REASONS[reason] || reason);
+  log.info("shutdown", REASONS[reason] || reason);
```

Comment:
Why do we need to distinguish all these types of shutdowns? Perhaps give an example for each `if` statement of what you might do differently in that case?

Reply:
I welcome specific language.

Reply:
Well that's the thing, as a new shield developer, I'm not sure myself what kinds of things I might want/need to do for each special case.